### PR TITLE
refactor(collate): deprecationwarning for `collate`

### LINKED ISSUE #459: Add deprecation notice for `collate`
In the future we plan to deprecate and remove `collate` functionality from Pycytominer. This issue describes the need to provide a deprecation notice to help warn users in advance of the functionality being removed so they may better prepare.

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This adds a DeprecationWarning to `collate` with guidance on replacements which may suit a person's possible needs. I did my best to suppress warnings when testing; one deprecation warning of about 8 total is still emitted though I'm not certain how. I opted to focus on getting the message in place here, as the warning here will also be removed eventually.

Thanks for any feedback!

Closes #459 

## What is the nature of your change?

- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

**Review by kenibrewer — APPROVED** [review]
LGTM! Thanks for adding!

File: pycytominer/cyto_utils/collate.py [hunk 1]
```diff
@@ -81,6 +82,16 @@ def collate(
             """
         )
 
+    # show a warning about collate deprecation
+    warnings.warn(
+        (
+            "pycytominer.cyto_utils.collate is deprecated and will be removed in future versions of Pycytominer. "
+            "Please consider using CellProfiler's ExportToDatabase module to create single-cell "
+            "profile SQLite files or CytoTable to create similar Parquet file output."
```
Comment (gwaybio) [inline pycytominer/cyto_utils/collate.py:90]:
```suggestion
            With the deprecation of cytominer-database, "pycytominer.cyto_utils.collate will be removed in future versions of Pycytominer. "
            "Please consider using CellProfiler's ExportToDatabase module to create single-cell "
            "SQLite files or CytoTable to create single-cell Parquet files."
```

**Review by gwaybio — APPROVED** [review]
👍

File: pycytominer/cyto_utils/collate.py (hunk 1 above)
Comment (d33bs) [reply pycytominer/cyto_utils/collate.py:90]: Thanks!

**Comment by d33bs** [comment]
Thanks @kenibrewer and @gwaybio ! Merging this in.

**Pull request merged**